core/app/models/spree/application_record.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Spree::ApplicationRecord < ActiveRecord::Base
```

Comment:
We might want to take into account, that the app itself has an `ApplicationRecord` as well?

```suggestion
base_class = defined?(ApplicationRecord) ? ApplicationRecord : ActiveRecord::Base
class Spree::ApplicationRecord < base_class
```

Reply:
Agree, but it could also happen that the app has an `ApplicationRecord` that should not be shared with Solidus' models maybe?

Reply:
Good point. It's hard to tell, because there might be code inside the ApplicationRecord that is related to Solidus' tables as well and there might be cases where this is not the case. I think this can be solved by subclassing ApplicationRecord in the app so those models have a seperate config.

Reply:
Thank you for the quick review 🙇 
> This is a good change, thanks 👏
> 
> But, I recommend to name not name this class `ApplicationRecord` even though it is namespaced it potentially conflicts with the top level `ApplicationRecord` new Rails apps have.
> 
> What about `Spree::BaseRecord`? Also, we already have `Spree::Base`, right? 
What do we do about it?

I think that is important to keep `ApplicationRecord` name because its has become a Rails convention, app and engine generators since Rails 5 create this class and almost everybody expects this name.
If we are not sure that namespacing is not enough we could use the `require_dependency` trick suggested here https://guides.rubyonrails.org/engines.html#app-directory

Reply:
I'm not convinced that `Spree::ApplicationRecord` should subclass the app one or vice-versa, one of the reasons why `ApplicationRecord` was introduced by Rails was to keep app models hierarchies separated from engines models hierarchies.
And the same goes for `ApplicationController`, `ApplicationMailer` and `ApplicationJob`, they all are [meant to be app or engine specific](https://guides.rubyonrails.org/engines.html#app-directory).

Reply:
I think you are right, this is the suggested way: in the documentation linked it states:

> The ArticlesController class inherits from Blorgh::ApplicationController, not the application's ApplicationController.